### PR TITLE
enhance: context menu on custom reports page

### PR DESCRIPTION
**Pull request by UnderKoen**

### CODE REVIEW

File: packages/desktop-client/src/components/reports/ReportCard.tsx [hunk 1]
```diff
@@ -141,32 +157,48 @@ function Layout({ children, isEditing, menuItems, onMenuSelect }: LayoutProps) {
...
-        >
-          <MenuButton ref={triggerRef} onPress={() => setMenuOpen(true)} />
+      {menuItems && (
+        <>
+          {isEditing && (
+            <View
+              className={[
+                menuOpen ? undefined : 'hover-visible',
+                NON_DRAGGABLE_AREA_CLASS_NAME,
+              ].join(' ')}
```
[the remark this answers is not in the document]
Comment (UnderKoen) [reply packages/desktop-client/src/components/reports/ReportCard.tsx:167]: This was not changed in this PR, and it was working before so this is a non issue

**Comment by youngcw** [comment]
This will probably need gated by the reports dashboard feature flag since it is using the dashboard features in the menu.

File: packages/desktop-client/src/components/reports/ReportCard.tsx [hunk 2]
```diff
@@ -119,10 +120,27 @@ type LayoutProps = {
 
 function Layout({ children, isEditing, menuItems, onMenuSelect }: LayoutProps) {
   const triggerRef = useRef(null);
-  const [menuOpen, setMenuOpen] = useState(false);
+  const viewRef = useRef(null);
+  const [menuOpen, setMenuOpen] = useState<null | 'context' | 'button'>(null);
+  const [crossOffset, setCrossOffset] = useState(0);
+  const [offset, setOffset] = useState(0);
+  const reportBudgetEnabled = useFeatureFlag('reportBudget');
```
Comment (matt-fidd) [inline packages/desktop-client/src/components/reports/ReportCard.tsx:127]: I don't think report budget is needed here, did you mean the dashboards feature flag?

File: upcoming-release-notes/3776.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [UnderKoen]
+---
+
+Add context menu's to custom reports dashboard
```
Comment (matt-fidd) [inline upcoming-release-notes/3776.md:6]:
```suggestion
Add context menus to custom reports dashboard
```

**Comment by matt-fidd** [comment]
This will need altering a bit now that dashboards are released, there's no need for the feature flag anymore

**Comment by matt-fidd** [comment]
I've found a bug:
1. Right click report
2. Rename
3. Click inside the name input box (as if you were placing the cursor)
4. The report opens instead of placing the cursor

**Review by MikesGlitch — APPROVED** [review]
Nice!

**Review by matt-fidd — APPROVED** [review]
Hadn't seen the change, sorry! LGTM :)

**Pull request merged**